File: backend/app/models/text_models.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class TextRequest:
    """Text processing request model"""
    text: str
    max_length: Optional[int] = 150
    min_length: Optional[int] = 30
    variations: Optional[int] = 1
    
    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> 'TextRequest':
        return cls(
            text=data.get('text', ''),
            max_length=data.get('max_length', 150),
            min_length=data.get('min_length', 30),
            variations=data.get('variations', 1)
        )

@dataclass
class TextResponse:
    """Text processing response model"""
    success: bool
    original_text: str
    processed_text: str
    processing_time: float
    word_count_original: int
    word_count_processed: int
    compression_ratio: Optional[float] = None
    variations: Optional[List[str]] = None
    error: Optional[str] = None
    
    def to_dict(self) -> Dict[str, Any]:
        result = {
            'success': self.success,
            'original_text': self.original_text,
            'processed_text': self.processed_text,
            'processing_time': round(self.processing_time, 3),
            'word_count_original': self.word_count_original,
            'word_count_processed': self.word_count_processed
        }
        
        if self.compression_ratio is not None:
            result['compression_ratio'] = round(self.compression_ratio, 2)
        
        if self.variations:
            result['variations'] = self.variations
        
        if self.error:
            result['error'] = self.error
            
        return result

@dataclass
class TextAnalysis:
    """Text analysis result model"""
    word_count: int
    sentence_count: int
    paragraph_count: int
    character_count: int
    character_count_no_spaces: int
    reading_time_minutes: float
    readability_score: float
    sentiment_score: float
    sentiment_label: str
    
    def to_dict(self) -> Dict[str, Any]:
        return {
            'word_count': self.word_count,
            'sentence_count': self.sentence_count,
            'paragraph_count': self.paragraph_count,
            'character_count': self.character_count,
            'character_count_no_spaces': self.character_count_no_spaces,
            'reading_time_minutes': round(self.reading_time_minutes, 1),
            'readability_score': round(self.readability_score, 1),
            'sentiment': {
                'score': round(self.sentiment_score, 3),
                'label': self.sentiment_label
            }
        }

File: lib/models/text_models.dart
class TextRequest {
  final String text;
  final int? maxLength;
  final int? minLength;
  final int? variations;

  const TextRequest({
    required this.text,
    this.maxLength,
    this.minLength,
    this.variations,
  });

  Map<String, dynamic> toJson() {
    return {
      'text': text,
      if (maxLength != null) 'max_length': maxLength,
      if (minLength != null) 'min_length': minLength,
      if (variations != null) 'variations': variations,
    };
  }

  factory TextRequest.fromJson(Map<String, dynamic> json) {
    return TextRequest(
      text: json['text'] ?? '',
      maxLength: json['max_length'],
      minLength: json['min_length'],
      variations: json['variations'],
    );
  }
}

class TextResponse {
  final bool success;
  final String? originalText;
  final String? processedText;
  final double? processingTime;
  final int? wordCountOriginal;
  final int? wordCountProcessed;
  final double? compressionRatio;
  final List<String>? variations;
  final String? error;

  const TextResponse({
    required this.success,
    this.originalText,
    this.processedText,
    this.processingTime,
    this.wordCountOriginal,
    this.wordCountProcessed,
    this.compressionRatio,
    this.variations,
    this.error,
  });

  factory TextResponse.fromJson(Map<String, dynamic> json) {
    return TextResponse(
      success: json['success'] ?? false,
      originalText: json['original_text'],
      processedText: json['processed_text'],
      processingTime: json['processing_time']?.toDouble(),
      wordCountOriginal: json['word_count_original'],
      wordCountProcessed: json['word_count_processed'],
      compressionRatio: json['compression_ratio']?.toDouble(),
      variations: json['variations'] != null 
          ? List<String>.from(json['variations'])
          : null,
      error: json['error'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'success': success,
      if (originalText != null) 'original_text': originalText,
      if (processedText != null) 'processed_text': processedText,
      if (processingTime != null) 'processing_time': processingTime,
      if (wordCountOriginal != null) 'word_count_original': wordCountOriginal,
      if (wordCountProcessed != null) 'word_count_processed': wordCountProcessed,
      if (compressionRatio != null) 'compression_ratio': compressionRatio,
      if (variations != null) 'variations': variations,
      if (error != null) 'error': error,
    };
  }
}